### PR TITLE
[ie/Picarto] Fix failed to parse JSON error

### LINKED ISSUE #10413: [Picarto] Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 2 column 1 (char 1)"))
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Provide a description that is worded well enough to be understood

Picarto and PicartoVod extractors are failing with this error: `Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 2 column 1 (char 1)"))`. The complete verbose output is below.

Picarto streams work in the browser, and yt-dlp works if provided the direct m3u8 stream URL that can be retrieved from Picarto's Debug Info.

Example live stream that I tried to download and failed: https://picarto.tv/PaintrFiend

Example VOD that I tried to download and failed: https://picarto.tv/ArtofZod/videos/578752

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU 'https://picarto.tv/PaintrFiend'
[debug] Command-line config: ['-vU', 'https://picarto.tv/PaintrFiend']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.07.09 from yt-dlp/yt-dlp [7ead7332a] (zip)
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: certifi-2024.07.04, secretstorage-3.3.1, sqlite3-3.37.2
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.07.09 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.07.09 from yt-dlp/yt-dlp)
[Picarto] Extracting URL: https://picarto.tv/PaintrFiend
[Picarto] PaintrFiend: Downloading JSON metadata
ERROR: [Picarto] PaintrFiend: PaintrFiend: Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 2 column 1 (char 1)")); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 740, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/picarto.py", line 32, in _real_extract
    data = self._download_json(
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1139, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1106, in download_handle
    return parse(self, content, video_id, transform_source=transform_source, fatal=fatal, errnote=errnote), urlh
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1094, in parse
    return getattr(ie, parser)(content, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1081, in _parse_json
    self.__print_error('Failed to parse JSON' if errnote is None else errnote, fatal, video_id, ve)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1064, in __print_error
    raise ExtractorError(f'{video_id}: {errnote}', cause=err)

  File "/usr/local/bin/yt-dlp/yt_dlp/utils/_utils.py", line 552, in decode
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 2 column 1 (char 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1078, in _parse_json
    return json.loads(
  File "/usr/lib/python3.10/json/__init__.py", line 359, in loads
    return cls(**kw).decode(s)
  File "/usr/local/bin/yt-dlp/yt_dlp/utils/_utils.py", line 560, in decode
json.decoder.JSONDecodeError: Expecting value in '': line 2 column 1 (char 1)
```

### PR DESCRIPTION
**Pull request by Frankgoji**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes the error that yt-dlp has been having for live streams and VODs: `Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 2 column 1 (char 1)"))`. It turns out that the requests to ptvintern.picarto.tv/ptvapi were getting back a web page with "Welcome to Apollo Server", rather than the expected JSON response, thus causing the JSON decode error. Luckily, by updating the headers in the request, the API returns the correct JSON responses.

Fixes #10413

### Testing

For VODs, the extractor test can be run to verify it now works:
```
$ hatch test PicartoVodIE_1
Running ['pytest', '-Werror', '--tb=short', '-p', 'no:randomly', 'test/test_download.py::TestDownl
oad::test_PicartoVod_1']
====================================== test session starts =======================================
platform linux -- Python 3.10.12, pytest-8.2.2, pluggy-1.5.0 -- /home/frankgoji/.local/share/hatch
/env/virtual/yt-dlp/NOjXHC0n/hatch-test.py3.10/bin/python3
cachedir: .pytest_cache
rootdir: /home/frankgoji/Documents/yt-dlp
configfile: pyproject.toml
plugins: rerunfailures-14.0, xdist-3.6.1
collected 1 item                                                                                 

test/test_download.py::TestDownload::test_PicartoVod_1 PASSED                              [100%]

======================================= 1 passed in 15.49s =======================================
```

For live streams, I made a binary with my changes with `make yt-dlp` and ran the local version to verify it was no longer having the JSON decode error:
```
$ ./yt-dlp --skip-download -v 'https://picarto.tv/PaintrFiend'
[debug] Command-line config: ['--skip-download', '-v', 'https://picarto.tv/PaintrFiend']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.07.02 from yt-dlp/yt-dlp [93d33cb29] (zip)
[debug] Lazy loading extractors is disabled
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-wi
th-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: certifi-2024.07.04, secretstorage-3.3.1, sqlite3-3.37.2
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1831 extractors
[Picarto] Extracting URL: https://picarto.tv/PaintrFiend
[Picarto] PaintrFiend: Downloading JSON metadata
[Picarto] PaintrFiend: Downloading load balancing info
[Picarto] PaintrFiend: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), 
channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: best/bestvideo+bestaudio
[info] PaintrFiend: Downloading 1 format(s): hls-3164
```

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/picarto.py [hunk 1]
```diff
@@ -43,16 +43,24 @@ def _real_extract(self, url):
     url
   }
 }''' % (channel_id, channel_id),  # noqa: UP031
-            })['data']
+            },
+            headers={
+                'Accept': '*/*',
+                'Content-Type': 'application/json',
+            },
+        )['data']
```
Comment (bashonly) [inline yt_dlp/extractor/picarto.py:51]:
```suggestion
            }, headers={'Accept': '*/*', 'Content-Type': 'application/json'})['data']
```

File: yt_dlp/extractor/picarto.py [hunk 2]
```diff
@@ -43,16 +43,24 @@ def _real_extract(self, url):
...
         title = metadata['title']
 
+        url = data['getLoadBalancerUrl']['url']
+        https_url = url.replace('http', 'https', 1) if url.startswith('http:') else url
         cdn_data = self._download_json(
-            data['getLoadBalancerUrl']['url'] + '/stream/json_' + metadata['stream_name'] + '.js',
-            channel_id, 'Downloading load balancing info')
+            https_url + '/stream/json_' + metadata['stream_name'] + '.js',
+            channel_id, 'Downloading load balancing info',
+        )
```
Comment (bashonly) [inline yt_dlp/extractor/picarto.py:63]: I would import `update_url` from utils and do this:
```py
        cdn_data = self._download_json(''.join((
            update_url(data['getLoadBalancerUrl']['url'], scheme='https'),
            '/stream/json_', metadata['stream_name'], '.js')),
            channel_id, 'Downloading load balancing info')
```

File: yt_dlp/extractor/picarto.py [hunk 3]
```diff
@@ -131,7 +139,12 @@ def _real_extract(self, url):
     }}
   }}
 }}''',
-            })['data']['video']
+            },
+            headers={
+                'Accept': '*/*',
+                'Content-Type': 'application/json',
+            },
+        )['data']['video']
```
Comment (bashonly) [inline yt_dlp/extractor/picarto.py:147]:
```suggestion
            }, headers={'Accept': '*/*', 'Content-Type': 'application/json'})['data']['video']
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/picarto.py (hunk 1 above)
Comment (Frankgoji) [reply yt_dlp/extractor/picarto.py]: Thanks, fixed.

File: yt_dlp/extractor/picarto.py (hunk 2 above)
Comment (Frankgoji) [reply yt_dlp/extractor/picarto.py]: Thanks! Fixed this.

File: yt_dlp/extractor/picarto.py (hunk 3 above)
Comment (Frankgoji) [reply yt_dlp/extractor/picarto.py]: Thanks, fixed.

**Review by bashonly — APPROVED** [review]

**Pull request merged**